src/bbmustache.erl

```diff
@@ -689,3 +707,73 @@ is_recursive_data(_)                                -> false.
+    {ok, AppConfig} = application:get_all_key(bbmustache),
+    Version = proplists:get_value(vsn, AppConfig),
+    io:format("~p~n", [Version]);
+process_commands([commands | _], _Options) ->
+    Commands = "render [file,...]",
+    io:format("~s~n", [Commands]);
+process_commands(Options, ["render" | TemplateFileNames]) ->
+    KeyType = proplists:get_value(key_type, Options, atom),
+    RenderOptions = [{key_type, KeyType}],
+    DataFileName = proplists:get_value(data_file, Options),
+    Data = merge_data_files(DataFileName),
+    lists:foreach(fun(TemplateFileName) ->
+                        {ok, Template} = file:read_file(TemplateFileName),
+                        Ret = render(Template, Data, RenderOptions),
+                        io:format("~s~n", [Ret])
+                  end, TemplateFileNames);
+process_commands(_, _) ->
+    throw(unrecognized_or_unimplemented_command).
+
+%% @doc Returns command-line options.
+-spec option_spec_list() -> [getopt:option_spec()].
+option_spec_list() ->
+[
+    %% {Name,   ShortOpt,   LongOpt,        ArgSpec,        HelpMsg}
+    {help,      $h,         "help",         undefined,      "Show this help information."},
+    {version,   $v,         "version",      undefined,      "Output the current bbmustache version"},
+    {commands,  undefined,  "commands",     undefined,      "Show available commands."},
+    {key_type,  $k,         "key-type",     atom,           "Key type (atom | binary | string)."},
+    {data_file, $d,         "data-file",    string,         "Erlang terms file."}
+].
+
+%% @doc Returns command-line help.
+-spec help() -> ok.
+help() ->
+    getopt:usage(option_spec_list(), escript:script_name()).
+
+% @doc Checks for terms inclusing in terms using format "file".
+-spec check_terms_inclusion(string(), [term()], [term()]) -> [term()].
+check_terms_inclusion(RelativeRoot, [FileName | T], Terms) when is_list(FileName) ->
+    IncludedTerms = merge_data_files(filename:join(RelativeRoot, FileName)),
+    check_terms_inclusion(RelativeRoot, T, IncludedTerms ++ Terms);
+check_terms_inclusion(RelativeRoot, [Tuple | T], Terms) ->
+    check_terms_inclusion(RelativeRoot, T, [Tuple | Terms]);
+check_terms_inclusion(_RelativeRoot, [], Terms) ->
+    Terms.
+
+%% @doc Implements data file inclusing (see check_terms_inclusion/1).
+-spec merge_data_files(string()) -> [term()].
+merge_data_files(FileName) ->
+    case file:consult(FileName) of
+        {ok, Terms} ->
+            RelativeRoot = filename:dirname(FileName),
+            NewTerms = check_terms_inclusion(RelativeRoot, Terms, []),
+            %% Remove already defined variables from Acc,
+            %% append NewTerms, preserving order
+            lists:foldl(fun(NewTerm, A) ->
+                            lists:keydelete(element(1, NewTerm), 1, A)
+                        end, [], NewTerms) ++ NewTerms;
+        {error, Reason} ->
+            throw({unable_to_read_datafile, Reason})
```

Comment:
Always use `throw(string())`, and catch in main and output to standard error.

```erl
main(_) ->
    try case getopt:parse(option_spec_list(), Args) of
        _Ret -> ....
    of catch
       throw:Reason ->
            _ = io:format(standard_error, Reason),
           halt(1)
    end
end
```

Reply:
f551d29a39d620da2886959b5731a2da1e5481c3